includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1486,15 +1488,21 @@ public function process_refund( $order_id, $amount = null, $reason = '' ) {
 			// If the payment method is Interac, the refund already exists (refunded via Mobile app).
 			$is_refunded_off_session = Payment_Method::INTERAC_PRESENT === $this->get_payment_method_type_for_order( $order );
 			if ( $is_refunded_off_session ) {
-				$refund_amount = WC_Payments_Utils::prepare_amount( $amount ?? $order->get_total(), $order->get_currency() );
-				$refunds       = array_filter(
-					$this->payments_api_client->list_refunds( $charge_id )['data'],
+				$refund_amount              = WC_Payments_Utils::prepare_amount( $amount ?? $order->get_total(), $order->get_currency() );
+				$list_charge_refund_request = List_Charge_Refunds::create();
+				$list_charge_refund_request->set_charge( $charge_id );
+
+				$refunds = $list_charge_refund_request->send( 'wcpay_list_charge_refunds_request' );
+
+				$refunds = array_filter(
+					$refunds['data'] ?? [],
```

Comment:
I think we may avoid overriding `$refunds` here, and clarify the difference between them. 

Note that: 
- Before applying `array_filter`, `$refunds` an object of `WCPay\Core\Server\Response`, which implements ArrayAccess. 
- After that, it's a real `array`.

The new name is just a suggestion, you might choose a better and shorter one.

```suggestion
				$list_charge_refund_response = $list_charge_refund_request->send( 'wcpay_list_charge_refunds_request' );

				$refunds = array_filter(
					$list_charge_refund_response['data'] ?? [],
```

Reply:
I wanted to save some memory. We don't use data we got originally got from the request because it is parsed in this filter function. Overriding `$refunds` with new data will prevent creating a new variable and memory allocation for the same.

Reply:
Gotcha, however, the memory is not a big deal here while using my approach is giving future readers better readability.